R/checks.R

```diff
@@ -110,48 +110,24 @@ check_coverages <- function(wcs, coverages) {
   }
 }
 
-check_cov_contains_bbox <- function(summary, bbox, crs = NULL) {
+check_cov_contains_bbox <- function(summary, bbox, crs) {
   if (is.null(bbox)) {
     return()
   }
-  cov_bbox <- emdn_get_bbox(summary)
 
-  if (!is.null(crs)) {
-    bbox <- sf::st_bbox(bbox, crs = sf::st_crs(crs))
+  cov_bbox <- emdn_get_bbox(summary) |>
+    sf::st_as_sfc()
 
-    if (sf::st_crs(cov_bbox) != sf::st_crs(bbox)) {
-      bbox <- bbox |>
-        sf::st_as_sfc() |>
-        sf::st_transform(crs = sf::st_crs(cov_bbox)) |>
-        sf::st_bbox()
-    }
-  }
-  test_bbox <- !c(
-    bbox[c("xmax", "ymax")] <= cov_bbox[c("xmax", "ymax")],
-    bbox[c("xmin", "ymin")] <= cov_bbox[c("xmax", "ymax")],
-    bbox[c("xmin", "ymin")] >= cov_bbox[c("xmin", "ymin")],
-    bbox[c("xmax", "ymax")] >= cov_bbox[c("xmin", "ymin")]
-  )
+  bbox <- sf::st_bbox(bbox, crs = sf::st_crs(crs)) |>
+    sf::st_as_sfc() |>
+    sf::st_transform(crs = sf::st_crs(cov_bbox)) # cov_bbox can be the same
 
-  outlying_edges <- unique(names(test_bbox)[test_bbox])
-  if (length(outlying_edges) == 0L) {
-    outlying_edges <- ""
-  }
+  intersects <- isTRUE(sf::st_intersects(bbox, cov_bbox))
 
-  if (
-    all(test_bbox) ||
-      all(outlying_edges %in% c("ymax", "ymin")) ||
-      all(outlying_edges %in% c("xmax", "xmin"))
-  ) {
-    cli::cli_abort(
-      "{.var bbox} boundaries {.val {names(test_bbox)[test_bbox]}} lie
-            outside coverage extent. No overlapping data to download."
-    )
-  }
-  if (any(test_bbox)) {
+  if (!intersects) {
     cli::cli_warn(
-      "{.var bbox} boundaries {.val {names(test_bbox)[test_bbox]}} lie
-            outside coverage extent. No overlapping data to download."
+      "{.var bbox} boundaries lie outside coverage extent. 
```

Comment:
we could make this a message instead, that'd be shut up with the `"emodnet.wcs.quiet"` option. After all if you get no data for a given bounding box, supposedly you'd notice it anyway?